src/test/datascience/kernel-launcher/kernelFinder.vscode.test.node.ts

```diff
@@ -118,11 +134,21 @@ suite('DataScience - Kernels Finder', () => {
         ) as LocalKernelConnectionMetadata;
         assert.ok(juliaKernelSpec);
 
-        const kernelSpec = (await kernelFinder.findKernel(resourceToUse, {
-            kernelspec: juliaKernelSpec?.kernelSpec as any,
-            orig_nbformat: 4
-        })) as LocalKernelConnectionMetadata;
+        const kernelSpec = takeTopRankKernel(
+            await kernelFinder.rankKernels(resourceToUse, {
+                kernelspec: juliaKernelSpec?.kernelSpec as any,
+                orig_nbformat: 4
+            })
+        ) as LocalKernelConnectionMetadata;
         assert.ok(kernelSpec.kernelSpec);
         assert.deepEqual(kernelSpec.kernelSpec.name, juliaKernelSpec.kernelSpec.name);
     });
 });
+
+function takeTopRankKernel(
```

Comment:
Function to keep the old tests checking what they were checking. I also added some lower level exactMatch unit tests.